test-it/ViaTests/JSReferenceArraySearch.via

```diff
@@ -0,0 +1,72 @@
+// Autogenerated Vireo assembly file
+// Generated on 10/25/2019 14:25:10
+//---------------------------------------------------
+//VI Definition: ::Web Server::Interactive::Application::JSReferenceArraySearch.gviweb
+define (Application%3A%3AJSReferenceArraySearch%2Egviweb dv(.VirtualInstrument (
+    Locals: c(   // Data Space
+        e(dv(.String 'Application::JSReferenceArraySearch.gviweb')local0)
+        ce(dv(.Int32 0)c1)
+        ce(dv(.ErrorCluster (false 0 '' ))c2)
+        ce(dv(.Int32 2)c3)
+        e(.JavaScriptDynamicRefNum local4)
+        e(.ErrorCluster local5)
+        e(.JavaScriptDynamicRefNum local6)
+        e(.ErrorCluster local7)
+        e(.JavaScriptDynamicRefNum local8)
+        e(.ErrorCluster local9)
+        e(.Boolean local10)
+        de(dv(.Boolean false)dataItem_TestResult_NoError)
+        e(.Boolean local12)
+        e(a(.JavaScriptDynamicRefNum *) local13)
+        e(.Int32 local14)
+        de(dv(.Int32 0)dataItem_IndexOfElement)
+        e(.Int32 local16)
+        e(.Boolean local17)
+        de(dv(.Boolean false)dataItem_TestResult_CorrectIndex)
+        e(.Boolean local19)
+    )
+        clump(1
+        Application%3A%3ACatalog%2Ejsli%3A%3ANow(c2 local4 local5)
+        Application%3A%3ACatalog%2Ejsli%3A%3ANow(local5 local6 local7)
+        Application%3A%3ACatalog%2Ejsli%3A%3ANow(local7 local8 local9)
+        IsEQ(local9 c2 local10)
+        Copy(local10 dataItem_TestResult_NoError)
+        ArrayConcatenate(local13 local4 local6 local8 )
+        Search1DArray(local13 local8 c1 local14 )
+        Copy(local14 dataItem_IndexOfElement)
+        IsEQ(local14 c3 local17)
+        Copy(local17 dataItem_TestResult_CorrectIndex)
+        Printf ("dataItem_IndexOfElement: %z\n" dataItem_IndexOfElement)
+        Printf ("dataItem_TestResult_CorrectIndex: %z\n" dataItem_TestResult_CorrectIndex)
+        Printf ("dataItem_TestResult_NoError: %z\n" dataItem_TestResult_NoError)
```

Comment:
Please consider adding the following test cases:
1.  Call "Search 1D Array" with a valid JS Reference returned by a previous JavaScriptInvoke call as "element" but do not include in the input "1D array".  Return -1?
2.  Call "Search 1D Array" with a variable of type JS Reference but not used at all (so it is Invalid JSRef) as "element" and that is not in the input "1D array". Return -1?
3.  Call "Search 1D Array" with a variable of type JS Reference but not used at all (so it is Invalid JSRef) as "element" and include this invalid JS Ref also in the input "1D array". Return correct index?

Reply:
Added test cases as requested:

![JSReferenceArraySearh](https://user-images.githubusercontent.com/8452083/67811273-676a2e00-fa6a-11e9-8fd5-336c96c48a2a.png)